modApiServer/src/org/aion/api/server/zmq/ProtocolProcessor.java

```diff
@@ -53,24 +61,41 @@
 public class ProtocolProcessor implements Runnable {
 
     protected static final Logger LOG = AionLoggerFactory.getLogger(LogEnum.API.name());
-    private final IHdlr handler;
     private static final String AION_ZMQ_WK_TH = "inproc://aionZmqWkTh";
     private static final String AION_ZMQ_CB_TH = "inproc://aionZmqCbTh";
     private static final String AION_ZMQ_EV_TH = "inproc://aionZmqEvTh";
     private static final String AION_ZMQ_HB_TH = "inproc://aionZmqHbTh";
-
-    private CfgApiZmq cfgApi;
-    private AtomicBoolean shutDown = new AtomicBoolean();
-
+    private static final String CURVEKEY_PATH;
+    private static final Path PATH;
     private static final long zmqHWM = 100_000;
     private static final int SOCKETID_LEN = 5;
     private static final int SOCKET_RECV_TIMEOUT = 3000;
 
+    static {
+        String storageDir = System.getProperty("local.storage.dir");
```

Comment:
Can we do the following to avoid the reliance on the static block:

- make it so that `storageDir` is an argument to the constructor
- make `CURVEKEY_PATH` and `PATH` non-static
- move the logic of setting CURVEKEY_PATH and PATH into constructor
- add either another constructor or a static method, that will call the constructor, and pass in `System.getProperty('local.storage.dir') as the value for storageDir